src/basics.md

```diff
@@ -1188,6 +1189,48 @@ Error level - description
 
 Run the recipe with `RUST_BACKTRACE=1` to display a detailed [`backtrace`] associated with this error.
 
+[ex-measure-elapsed-time]: #ex-measure-elapsed-time
+<a name="ex-measure-elapsed-time"></a>
+## Measure the elapsed time between two code sections
+
+[![std-badge]][std] [![cat-time-badge]][cat-time]
+
+One way to get the elapsed time is to directly use the method 
+[`time::Instant::elapsed`].
+
+```rust
+use std::time::{Duration, Instant};
+use std::thread;
+
+fn expensive_function() {
+    thread::sleep(Duration::from_millis(10));
+}
+
+fn main() {
+    let start = Instant::now();
+    expensive_function();
+    let duration = start.elapsed();
+    
+    println!("Time elapsed in expensive_function() is: {:?}", duration);
```

Comment:
One possible tripwire here is that `elapsed` does not reset the start in any way.

We can suggest following or just mention the fact in the description.

```rust
let start = Instant::now();
expensive_function();
let (start, duration) = (Instant::now(), start.elapsed());
println!("Time elapsed in expensive_function() is: {:?}", duration);
some_other_function();
println!("Time elapsed in some_other_function() is: {:?}", start.elapsed());
```

Reply:
I will just mention it in the description since I think this example will be a bit harder to follow and since start is reset before the println!(), makes it a bit imprecise.